src/config.c

```diff
@@ -628,6 +628,9 @@ void loadServerConfigFromString(char *config) {
     if (server.config_hz < CONFIG_MIN_HZ) server.config_hz = CONFIG_MIN_HZ;
     if (server.config_hz > CONFIG_MAX_HZ) server.config_hz = CONFIG_MAX_HZ;
 
+    /* For backward compatibility with the Redis OSS */
+    server.syslog_ident = server.extended_redis_compat ? "redis" : SERVER_NAME;
```

Comment:
This looks like it's overriding any value configured by the user.

Reply:
Yes, you are right. It is little bit tricky to gate behind the redis-compat config than I thought before :) 
Anyway, I will fix this problem.

Reply:
An idea: Maybe we can set the default value at `createStringConfig("syslog-ident", ...)` to NULL. Then, here, we can set it to "redis" or SERVER_NAME if it still has no value. I didn't try. It's just an idea.

```suggestion
    if (server.syslog_ident == NULL)
        server.syslog_ident = server.extended_redis_compat ? "redis" : SERVER_NAME;
```

Reply:
I think adding server.extended_redis_compat condition here makes thing complex. In other places, by adding server.extended_redis_compat, the idea is to makes output log compatible with Redis version.    However, considering the redis case: the only values for syslog-ident are redis and other value; and for valkey case: the only values for syslog-ident are valkey and other value.         And we can also find the config syslog-ident is IMMUTABLE, thus I think we do not need to touch other files, only update the default value to "valkey" in config.c is enough.  If the client wants to use "syslog-ident redis", the only thing to change is to add one line in the conf file "syslog-ident".

Reply:
@hwware Of course, you're right! A user can just set syslog-ident to "redis".

By brain was overcomplicating it.

Reply:
> An idea: Maybe we can set the default value at `createStringConfig("syslog-ident", ...)` to NULL. 

Thanks for the suggestion @zuiderkwast. if we need to set it to NULL, then I realized that we also need to change the `ALLOW_EMPTY_STRING` to `EMPTY_STRING_IS_NULL`. But, then it will change the existing  behavior where if user provided empty-string for the syslog-ident can also be changed to "redis" or "valkey" depending on the extended compatibility config.

Reply:
@karthyuom Thanks for solving this fancy logic, but I actually think the simple solution suggested by @hwware is better. What matters is that it's possible to configure it to be redis compatible.

We don't want to add unnecessary complexity for this temporary compatibility-config that we will remove in a few versions anyway.

Reply:
@zuiderkwast Yes I agree. Then I can revert back to the original commit where user can even configure "redis" for backward compatibility. Thanks @hwware for bringing this up.

Reply:
Yes just use SERVER_NAME for the default value.